org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/RecordPatternTest.java

```diff
@@ -2008,8 +2008,8 @@ public void testRecordPatternTypeInference_011() {
 				"----------\n" +
 				"1. ERROR in X.java (at line 10)\n" +
 				"	if (p instanceof R<>(String a)) {\n" +
-				"	    ^^^^^^^^^^^^^^^^^^^^^^^^^^\n" +
-				"Incompatible conditional operand types R<capture#1-of ? extends I> and R\n" +
+				"	                 ^^^^^^^^^^^^^\n" +
+				"Type mismatch: cannot convert from R<capture#1-of ? extends I> to R\n" +
```

Comment:
This test doesn't compile with javac so unable to verify behavior - bug in javac 

I get:

```
X.java:10: error: illegal start of type
        if (p instanceof R<>(String a)) {
                           ^
1 error

```

Reply:
Good catch. To be continued in #3157.